test/Confluent.Kafka.IntegrationTests/Tests/Builder_CustomDefaults.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2016-2019 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+#pragma warning disable xUnit1026
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Xunit;
+
+
+namespace Confluent.Kafka.IntegrationTests
+{
+    /// <summary>
+    ///     Create a custom builder for producer / consumer that uses
+    ///     different default serdes for string. This test is primarily
+    ///     to demonstrate the suitability of the API for this purpose.
+    /// </summary>
+    public static partial class Tests
+    {
+        class MyProducerBuilder<K, V> : ProducerBuilder<K, V>
+        {
+            class Utf32Serializer : ISerializer<string>
+            {
+                public byte[] Serialize(string data, bool isKey, MessageMetadata messageMetadata, TopicPartition destination)
```

Comment:
Is the partition relevant here? In most cases it will not be set, since partitioning is done after serialization, right?

Reply:
this is why we're getting you to review these PRs :-).

I will fix this in another PR after this one is merged - it's not directly related to this PR - is a result of an earlier mistake. There is also the question of whether partition should be on the deserializer. probably not, probably should just be topic.